kerneldriver_linux.go

```diff
@@ -0,0 +1,68 @@
+// +build linux
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// File originally made by jpoirier (https://github.com/jpoirier)
+// Addapted to the google/gousb library by nicovell3
+
+package gousb
+
+import (
+	"fmt"
+)
+
+/*
+#cgo pkg-config: libusb-1.0
+#include <libusb.h>
+*/
+import "C"
+
+// detachKernelDriver detaches any active kernel drivers, if supported by the platform.
+// If there are any errors, like Context.ListDevices, only the final one will be returned.
+func (d *Device) detachKernelDriver() (err error) {
```

Comment:
all of the C-related interfaces need to be in the libusbImpl in libusb.go, with appropriate modifications of the libusbIntf to match and a matching fakeLibusb modifications in fakelibusb_test.go. It's not possible to write tests otherwise.